Index: gcc/config/avr/avr.h
===================================================================
--- gcc/config/avr/avr.h	(revision 129539)
+++ gcc/config/avr/avr.h	(working copy)
@@ -737,7 +737,16 @@
 #define ASM_SPEC "%{mmcu=avr25:-mmcu=avr2;\
 mmcu=*:-mmcu=%*}"
 
-#define LINK_SPEC " %{!mmcu*:-m avr2}\
+#define LINK_SPEC "\
+%{mrelax:--relax\
+         %{mpmem-wrap-around:%{mmcu=at90usb8*:--pmem-wrap-around=8k}\
+                             %{mmcu=atmega16*:--pmem-wrap-around=16k}\
+                             %{mmcu=atmega32*|\
+                               mmcu=at90can32*:--pmem-wrap-around=32k}\
+                             %{mmcu=atmega64*|\
+                               mmcu=at90can64*|\
+                               mmcu=at90usb64*:--pmem-wrap-around=64k}}}\
+%{!mmcu*:-m avr2}\
 %{mmcu=at90s1200|\
   mmcu=attiny11|\
   mmcu=attiny12|\
