--- gcc/config/avr/avr.c.old	2007-09-26 10:04:29.856125000 -0600
+++ gcc/config/avr/avr.c	2007-09-26 10:31:39.028000000 -0600
@@ -218,6 +218,7 @@ static const struct mcu_type_s avr_mcu_t
   { "attiny261",    ARCH_AVR25, "__AVR_ATtiny261__" },
   { "attiny461",    ARCH_AVR25, "__AVR_ATtiny461__" },
   { "attiny861",    ARCH_AVR25, "__AVR_ATtiny861__" },
+  { "attiny43u",    ARCH_AVR25, "__AVR_ATtiny43U__" },
   { "at86rf401",    ARCH_AVR25, "__AVR_AT86RF401__" },
     /* Classic, > 8K.  */
   { "avr3",         ARCH_AVR3, NULL },
--- gcc/config/avr/avr.h.old	2007-09-26 10:31:54.840500000 -0600
+++ gcc/config/avr/avr.h	2007-09-26 10:33:50.762375000 -0600
@@ -858,6 +858,7 @@ mmcu=*:-mmcu=%*}"
 %{mmcu=attiny261:crttn261.o%s} \
 %{mmcu=attiny461:crttn461.o%s} \
 %{mmcu=attiny861:crttn861.o%s} \
+%{mmcu=attiny43u:crttn43u.o%s} \
 %{mmcu=atmega103|mmcu=avr3:crtm103.o%s} \
 %{mmcu=atmega603:crtm603.o%s} \
 %{mmcu=at43usb320:crt43320.o%s} \
--- gcc/config/avr/t-avr.old	2007-09-26 10:34:04.199875000 -0600
+++ gcc/config/avr/t-avr	2007-09-26 10:34:50.824875000 -0600
@@ -53,6 +53,7 @@ MULTILIB_MATCHES = \
 	mmcu?avr25=mmcu?attiny261 \
 	mmcu?avr25=mmcu?attiny461 \
 	mmcu?avr25=mmcu?attiny861 \
+	mmcu?avr25=mmcu?attiny43u \
 	mmcu?avr25=mmcu?at86rf401 \
 	mmcu?avr3=mmcu?atmega103 \
 	mmcu?avr3=mmcu?atmega603 \
