--- gcc/config/avr/avr.c.old	2007-10-07 20:11:40.071875000 -0600
+++ gcc/config/avr/avr.c	2007-10-07 20:12:38.103125000 -0600
@@ -219,6 +219,7 @@ static const struct mcu_type_s avr_mcu_t
   { "attiny461",    ARCH_AVR25, "__AVR_ATtiny461__" },
   { "attiny861",    ARCH_AVR25, "__AVR_ATtiny861__" },
   { "attiny43u",    ARCH_AVR25, "__AVR_ATtiny43U__" },
+  { "attiny48",     ARCH_AVR25, "__AVR_ATtiny48__" },
   { "at86rf401",    ARCH_AVR25, "__AVR_AT86RF401__" },
     /* Classic, > 8K.  */
   { "avr3",         ARCH_AVR3, NULL },
--- gcc/config/avr/avr.h.old	2007-10-07 20:11:45.462500000 -0600
+++ gcc/config/avr/avr.h	2007-10-07 20:29:43.040625000 -0600
@@ -806,6 +806,7 @@ mmcu=*:-mmcu=%*}"
   mmcu=atmega169*|\
   mmcu=atmega8hva|\
   mmcu=atmega16hva|\
+  mmcu=attiny48|\
   mmcu=at90can*|\
   mmcu=at90pwm*|\
   mmcu=at90usb*: -Tdata 0x800100}\
@@ -859,6 +860,7 @@ mmcu=*:-mmcu=%*}"
 %{mmcu=attiny461:crttn461.o%s} \
 %{mmcu=attiny861:crttn861.o%s} \
 %{mmcu=attiny43u:crttn43u.o%s} \
+%{mmcu=attiny48:crttn48.o%s} \
 %{mmcu=atmega103|mmcu=avr3:crtm103.o%s} \
 %{mmcu=atmega603:crtm603.o%s} \
 %{mmcu=at43usb320:crt43320.o%s} \
--- gcc/config/avr/t-avr.old	2007-10-07 20:11:56.290625000 -0600
+++ gcc/config/avr/t-avr	2007-10-07 20:31:22.384375000 -0600
@@ -54,6 +54,7 @@ MULTILIB_MATCHES = \
 	mmcu?avr25=mmcu?attiny461 \
 	mmcu?avr25=mmcu?attiny861 \
 	mmcu?avr25=mmcu?attiny43u \
+	mmcu?avr25=mmcu?attiny48 \
 	mmcu?avr25=mmcu?at86rf401 \
 	mmcu?avr3=mmcu?atmega103 \
 	mmcu?avr3=mmcu?atmega603 \
