--- gcc/config/avr/avr.c.old	2007-10-09 10:04:13.562500000 -0600
+++ gcc/config/avr/avr.c	2007-10-09 10:05:12.671875000 -0600
@@ -238,7 +238,9 @@ static const struct mcu_type_s avr_mcu_t
   { "atmega8hva",   ARCH_AVR4, "__AVR_ATmega8HVA__" },
   { "at90pwm1",     ARCH_AVR4, "__AVR_AT90PWM1__" },
   { "at90pwm2",     ARCH_AVR4, "__AVR_AT90PWM2__" },
+  { "at90pwm2b",    ARCH_AVR4, "__AVR_AT90PWM2B__" },
   { "at90pwm3",     ARCH_AVR4, "__AVR_AT90PWM3__" },
+  { "at90pwm3b",    ARCH_AVR4, "__AVR_AT90PWM3B__" },
     /* Enhanced, > 8K.  */
   { "avr5",         ARCH_AVR5, NULL },
   { "atmega16",     ARCH_AVR5, "__AVR_ATmega16__" },
--- gcc/config/avr/avr.h.old	2007-10-09 10:04:18.234375000 -0600
+++ gcc/config/avr/avr.h	2007-10-09 10:13:25.265625000 -0600
@@ -775,7 +775,9 @@ mmcu=*:-mmcu=%*}"
   mmcu=atmega48|\
   mmcu=at90pwm1|\
   mmcu=at90pwm2|\
-  mmcu=at90pwm3:-m avr4}\
+  mmcu=at90pwm2b|\
+  mmcu=at90pwm3|\
+  mmcu=at90pwm3b:-m avr4}\
 %{mmcu=atmega16*|\
   mmcu=atmega32*|\
   mmcu=atmega406|\
@@ -873,7 +875,9 @@ mmcu=*:-mmcu=%*}"
 %{mmcu=atmega8535:crtm8535.o%s} \
 %{mmcu=at90pwm1:crt90pwm1.o%s} \
 %{mmcu=at90pwm2:crt90pwm2.o%s} \
+%{mmcu=at90pwm2b:crt90pwm2b.o%s} \
 %{mmcu=at90pwm3:crt90pwm3.o%s} \
+%{mmcu=at90pwm3b:crt90pwm3b.o%s} \
 %{mmcu=atmega16:crtm16.o%s} \
 %{mmcu=atmega161|mmcu=avr5:crtm161.o%s} \
 %{mmcu=atmega162:crtm162.o%s} \
--- gcc/config/avr/t-avr.old	2007-10-09 10:04:25.921875000 -0600
+++ gcc/config/avr/t-avr	2007-10-09 10:14:35.375000000 -0600
@@ -69,7 +69,9 @@ MULTILIB_MATCHES = \
 	mmcu?avr4=mmcu?atmega8hva \
 	mmcu?avr4=mmcu?at90pwm1 \
 	mmcu?avr4=mmcu?at90pwm2 \
+	mmcu?avr4=mmcu?at90pwm2b \
 	mmcu?avr4=mmcu?at90pwm3 \
+	mmcu?avr4=mmcu?at90pwm3b \
 	mmcu?avr5=mmcu?atmega16 \
 	mmcu?avr5=mmcu?atmega161 \
 	mmcu?avr5=mmcu?atmega162 \
