--- gcc/config/avr/avr.c.orig	2007-11-29 15:11:34.151035300 -0700
+++ gcc/config/avr/avr.c	2007-11-29 15:16:50.295220100 -0700
@@ -149,6 +149,7 @@ static const struct base_arch_s avr_arch
   { 0, 0, 0, 0, 0, "__AVR_ARCH__=2" },
   { 0, 0, 0, 1, 0, "__AVR_ARCH__=25"},
   { 0, 0, 1, 0, 0, "__AVR_ARCH__=3" },
+  { 0, 0, 1, 1, 0, "__AVR_ARCH__=35"},
   { 0, 1, 0, 1, 0, "__AVR_ARCH__=4" },
   { 0, 1, 1, 1, 0, "__AVR_ARCH__=5" },
   { 0, 1, 1, 1, 1, "__AVR_ARCH__=6" }
@@ -164,6 +165,7 @@ enum avr_arch
   ARCH_AVR2,
   ARCH_AVR25,
   ARCH_AVR3,
+  ARCH_AVR35,
   ARCH_AVR4,
   ARCH_AVR5,
   ARCH_AVR6
@@ -222,6 +224,10 @@ static const struct mcu_type_s avr_mcu_t
   { "at43usb320",   ARCH_AVR3, "__AVR_AT43USB320__" },
   { "at43usb355",   ARCH_AVR3, "__AVR_AT43USB355__" },
   { "at76c711",     ARCH_AVR3, "__AVR_AT76C711__" },
+    /* Classic + MOVW + JMP/CALL.  */
+  { "avr35",        ARCH_AVR35, NULL },
+  { "at90usb82",    ARCH_AVR35, "__AVR_AT90USB82__" },
+  { "at90usb162",   ARCH_AVR35, "__AVR_AT90USB162__" },
     /* Enhanced, <= 8K.  */
   { "avr4",         ARCH_AVR4, NULL },
   { "atmega8",      ARCH_AVR4, "__AVR_ATmega8__" },
@@ -282,8 +288,6 @@ static const struct mcu_type_s avr_mcu_t
   { "at90can128",   ARCH_AVR5, "__AVR_AT90CAN128__" },
   { "at90pwm216",   ARCH_AVR5, "__AVR_AT90PWM216__" },
   { "at90pwm316",   ARCH_AVR5, "__AVR_AT90PWM316__" },
-  { "at90usb82",    ARCH_AVR5, "__AVR_AT90USB82__" },
-  { "at90usb162",   ARCH_AVR5, "__AVR_AT90USB162__" },
   { "at90usb646",   ARCH_AVR5, "__AVR_AT90USB646__" },
   { "at90usb647",   ARCH_AVR5, "__AVR_AT90USB647__" },
   { "at90usb1286",  ARCH_AVR5, "__AVR_AT90USB1286__" },
--- gcc/config/avr/t-avr.orig	2007-11-29 15:11:34.166661100 -0700
+++ gcc/config/avr/t-avr	2007-11-29 15:19:28.680949300 -0700
@@ -37,8 +37,8 @@ fp-bit.c: $(srcdir)/config/fp-bit.c $(sr
 
 FPBIT = fp-bit.c
 
-MULTILIB_OPTIONS = mmcu=avr2/mmcu=avr25/mmcu=avr3/mmcu=avr4/mmcu=avr5/mmcu=avr6
-MULTILIB_DIRNAMES = avr2 avr25 avr3 avr4 avr5 avr6
+MULTILIB_OPTIONS = mmcu=avr2/mmcu=avr25/mmcu=avr3/mmcu=avr35/mmcu=avr4/mmcu=avr5/mmcu=avr6
+MULTILIB_DIRNAMES = avr2 avr25 avr3 avr35 avr4 avr5 avr6
 
 # The many avr2 matches are not listed here - this is the default.
 MULTILIB_MATCHES = \
@@ -61,6 +61,8 @@ MULTILIB_MATCHES = \
 	mmcu?avr3=mmcu?at43usb320 \
 	mmcu?avr3=mmcu?at43usb355 \
 	mmcu?avr3=mmcu?at76c711 \
+	mmcu?avr35=mmcu?at90usb82 \
+	mmcu?avr35=mmcu?at90usb162 \
 	mmcu?avr4=mmcu?atmega48 \
 	mmcu?avr4=mmcu?atmega48p \
 	mmcu?avr4=mmcu?atmega8 \
@@ -117,8 +119,6 @@ MULTILIB_MATCHES = \
 	mmcu?avr5=mmcu?at90can128 \
     mmcu?avr5=mmcu?at90pwm216 \
     mmcu?avr5=mmcu?at90pwm316 \
-	mmcu?avr5=mmcu?at90usb82 \
-	mmcu?avr5=mmcu?at90usb162 \
 	mmcu?avr5=mmcu?at90usb646 \
 	mmcu?avr5=mmcu?at90usb647 \
 	mmcu?avr5=mmcu?at90usb1286 \
Index: gcc/config/avr/avr.h
===================================================================
--- gcc/config/avr/avr.h	(revision 130513)
+++ gcc/config/avr/avr.h	(working copy)
@@ -734,7 +734,7 @@
 /* A C string constant that tells the GCC driver program options to
    pass to `cc1plus'.  */
 
-#define ASM_SPEC "%{mmcu=avr25:-mmcu=avr2;\
+#define ASM_SPEC "%{mmcu=avr25:-mmcu=avr2;mmcu=avr35:-mmcu=avr3;\
 mmcu=*:-mmcu=%*}"
 
 #define LINK_SPEC " %{!mmcu*:-m avr2}\
@@ -759,7 +759,9 @@
   mmcu=attiny8*:-m avr2}\
 %{mmcu=atmega103|\
   mmcu=at43*|\
-  mmcu=at76*:-m avr3}\
+  mmcu=at76*|\
+  mmcu=at90usb82|\
+  mmcu=at90usb162:-m avr3}\
 %{mmcu=atmega8*|\
   mmcu=atmega48*|\
   mmcu=at90pwm1|\
@@ -775,7 +777,8 @@
   mmcu=at90can*|\
   mmcu=at90pwm216|\
   mmcu=at90pwm316|\
-  mmcu=at90usb*|\
+  mmcu=at90usb6*|\
+  mmcu=at90usb12*|\
   mmcu=at94k:-m avr5}\
 %{mmcu=atmega324*|\
   mmcu=atmega325*|\
